File: Python/projectEulerSolutionsTest.py
# https://projectbalint.com/en/page/project-euler-solutions/
#
# Test file for all Project Euler questions
# Copyright (c) 2018 Balint Gyevnar


import importlib
import time

available_solutions = {
    1: '233168',
    2: '4613732',
}

for k, v in available_solutions.items():
    module = importlib.import_module('p{}'.format(k))

    print('Problem {}: '.format(k), end='')

    start = time.time()
    answer = module.run()
    end = time.time() - start

    print('{0} - Elapsed time: {1:.8f}'.format(answer, end))

    assert v == answer

print('Tests ran successfuly')

File: CSharp/ProjectEulerSolutionsTest.cs
// ProjectEulerSolutionsTest.cs
//
// Test file for all PE questions
// Copyright (c) 2018 Balint Gyevnar

﻿using System;

namespace ProjectEulerSolutions
{
    public sealed class ProjectEulerSolutionsTest
    {
        // Testing all solutions
        public static void Main(string[] args)
        {
            AssertEquals("233168", new S1().Run());
            AssertEquals("4613732", new S2().Run());

            Console.WriteLine("All tests ran succesfuly.");
            Console.ReadLine();
        }

        // Simple method to test equality
        public static void AssertEquals(object objA, object objB)
        {
            if (!objA.Equals(objB))
                throw new Exception("Assertion failed");
        }
    }
}
